sklearn/tree/_splitter.pxd

```diff
@@ -46,13 +46,13 @@ cdef class Splitter:
     cdef object random_state             # Random state
     cdef UINT32_t rand_r_state           # sklearn_rand_r random number state
 
-    cdef SIZE_t* samples                 # Sample indices in X, y
+    cdef SIZE_t[::1] samples             # Sample indices in X, y
     cdef SIZE_t n_samples                # X.shape[0]
     cdef double weighted_n_samples       # Weighted number of samples
-    cdef SIZE_t* features                # Feature indices in X
-    cdef SIZE_t* constant_features       # Constant features indices
+    cdef SIZE_t[::1] features            # Feature indices in X
+    cdef SIZE_t[::1] constant_features   # Constant features indices
     cdef SIZE_t n_features               # X.shape[1]
-    cdef DTYPE_t* feature_values         # temp. array holding feature values
+    cdef DTYPE_t[::1] feature_values     # temp. array holding feature values
 
     cdef SIZE_t start                    # Start position for the current node
     cdef SIZE_t end                      # End position for the current node
```

Comment:
Do you plan to change as `sample_weight` in a future PR?

Reply:
What do you mean?

Reply:
Sorry I was referring to the pointer that is 2 lines below `sample_weight`.

Reply:
I think @glemaitre means:

https://github.com/scikit-learn/scikit-learn/blob/d247579341b22a8bbb765ee06716838140c6361c/sklearn/tree/_splitter.pxd#L61

Yes I plan to do it in the future. `sample_weight` touches multiple files, so I wanted to do it in another PR.

Reply:
Works with me.